tests/visualization_tests/test_pareto_front.py

```diff
@@ -1,451 +1,249 @@
-        )
 
-    with pytest.raises(ValueError, match=error_message):
-        plot_pareto_front(
-            study=study,
-            target_names=["Foo", "Bar", "Baz", "Qux"],
-            include_dominated_trials=include_dominated_trials,
-            axis_order=axis_order,
-            targets=targets,
-        )
-
-    target_names = ["Foo", "Bar", "Baz"]
-    figure = plot_pareto_front(study=study, target_names=target_names, axis_order=axis_order)
-    assert figure.layout.scene.xaxis.title.text == target_names[actual_axis_order[0]]
-    assert figure.layout.scene.yaxis.title.text == target_names[actual_axis_order[1]]
-    assert figure.layout.scene.zaxis.title.text == target_names[actual_axis_order[2]]
-
-
-@pytest.mark.parametrize("include_dominated_trials", [False, True])
-@pytest.mark.parametrize("use_constraints_func", [False, True])
-def test_plot_pareto_front_unsupported_dimensions(
-    include_dominated_trials: bool, use_constraints_func: bool
-) -> None:
-    constraints_func = (lambda _: [-1.0]) if use_constraints_func else None
-
-    error_message = (
-        "`plot_pareto_front` function only supports 2 or 3 objective"
-        " studies when using `targets` is `None`. Please use `targets`"
-        " if your objective studies have more than 3 objectives."
+    assert info == _ParetoFrontInfo(
+        n_targets=2,
+        target_names=target_names or ["Objective 0", "Objective 1"],
+        best_trials_with_values=[(trials[1], [1, 1]), (trials[2], [0, 2])],
+        non_best_trials_with_values=[(trials[0], [1, 2])] if include_dominated_trials else [],
+        infeasible_trials_with_values=[(trials[3], [1, 0])],
+        axis_order=axis_order or [0, 1],
+        include_dominated_trials=include_dominated_trials,
+        has_constraints_func=True,
     )
 
-    # Unsupported: n_objectives == 1.
-    with pytest.raises(ValueError, match=error_message):
-        study = optuna.create_study(directions=["minimize"])
-        study.optimize(lambda _: [0], n_trials=1)
-        plot_pareto_front(
-            study=study,
-            include_dominated_trials=include_dominated_trials,
-            constraints_func=constraints_func,
-        )
-
-    with pytest.raises(ValueError, match=error_message):
-        study = optuna.create_study(direction="minimize")
-        study.optimize(lambda _: [0], n_trials=1)
-        plot_pareto_front(
-            study=study,
-            include_dominated_trials=include_dominated_trials,
-            constraints_func=constraints_func,
-        )
-
-    # Unsupported: n_objectives == 4.
-    with pytest.raises(ValueError, match=error_message):
-        study = optuna.create_study(directions=["minimize", "minimize", "minimize", "minimize"])
-        study.optimize(lambda _: [0, 0, 0, 0], n_trials=1)
-        plot_pareto_front(
-            study=study,
-            include_dominated_trials=include_dominated_trials,
-            constraints_func=constraints_func,
-        )
```

Comment:
`_get_pareto_front_info` does not depend on the number of dimensions, so we do not need to check both 2D and 3D cases. We test both 2D and 3D cases of `_get_pareto_front_plot` in `test_get_pareto_front_plot`.